### PR TITLE
Added functionality for unsharp mask

### PR DESCRIPTION
**Pull request by AnimeshMaheshwari22**
### Description
Added the functionality for unsharp mask. I am extremely thankful to @edgarriba and @lferraz for their guidance. It is with their amazing suggestions that I have been able to add this functionality. Thankful to the Kornia community. The existing code has helped me learn a lot.

### CODE REVIEW

**Review by shijianjian — CHANGES_REQUESTED** [review]
tensor in and tensor out.

File: kornia/filters/__init__.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from .median import MedianBlur, median_blur
 from .motion import MotionBlur, MotionBlur3D, motion_blur, motion_blur3d
 from .filter import filter2D, filter3D
+from .unsharp import unsharp_mask, sharpen
```
Comment (lferraz) [inline kornia/filters/__init__.py:10]: why do you import sharpen?

File: kornia/filters/unsharp.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+from typing import Tuple
+import numpy as np
+import torch
+import torch.nn as nn
+
+import kornia
+from kornia.filters import GaussianBlur2d
+
+def sharpen(
+        input: np.ndarray,
```
Comment (lferraz) [inline kornia/filters/unsharp.py:10]: input/output must be tensors. Take into account that with the current code, you cannot use cuda.

File: kornia/filters/unsharp.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        kernel_size: Tuple[int, int],
+        sigma: Tuple[float, float],
+        border_type: str = 'reflect') -> np.ndarray:
+    r"""Creates an operator that blurs a tensor using the existing Gaussian filter available with the Kornia library.
+
+
```
Comment (lferraz) [inline kornia/filters/unsharp.py:16]: an stupid thing, typically it is a single new line required to separate the first line and the `Arguments:` section

File: kornia/filters/unsharp.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        <class 'numpy.ndarray'>
+    """
+    data: torch.tensor = kornia.image_to_tensor(input, keepdim=False)
+    data=data.float()/255.
+    gauss = kornia.filters.GaussianBlur2d(kernel_size, sigma)
+    data_blur: torch.tensor = gauss(data)
+    data_new = data + (data - data_blur)
+    img_new = kornia.tensor_to_image(data_new) * 255
+    return img_new
+
```
Comment (lferraz) [inline kornia/filters/unsharp.py:41]: another stupid thing, sorry, between classes typically there are only 2 new lines.

File: kornia/filters/unsharp.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        np.ndarray: the sharpened image.
+
+    Examples:
+        >>> input = cv2.imread(path_to_image)
+        >>> output = sharpen(input, (3, 3), (1.5, 1.5))
+        >>> type(output)
+        <class 'numpy.ndarray'>
+    """
+    data: torch.tensor = kornia.image_to_tensor(input, keepdim=False)
+    data=data.float()/255.
```
Comment (lferraz) [inline kornia/filters/unsharp.py:35]: probably this should be assumed once you change to an input tensor.

File: kornia/filters/unsharp.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    gauss = kornia.filters.GaussianBlur2d(kernel_size, sigma)
```
Comment (lferraz) [inline kornia/filters/unsharp.py:36]: gauss does not have typing

File: kornia/filters/unsharp.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    Returns:
+        np.ndarray: the sharpened image.
+
+    Examples:
+        >>> input = cv2.imread(path_to_image)
+        >>> output = sharpen(input, (3, 3), (1.5, 1.5))
+        >>> type(output)
+        <class 'numpy.ndarray'>
+    """
+    data: torch.tensor = kornia.image_to_tensor(input, keepdim=False)
```
Comment (lferraz) [inline kornia/filters/unsharp.py:34]: @edgarriba do we need to validate the input params?

File: test/filters/test_unsharp_mask.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import pytest
+
+import kornia
+import kornia.testing as utils  # test utils
+
+import torch
+from torch.autograd import gradcheck
+from torch.testing import assert_allclose
+
+class Testunsharp:
```
Comment (lferraz) [inline test/filters/test_unsharp_mask.py:10]: Maybe we should a numeric test, with an small predefined input tensor. @edgarriba do you think it could be interesting?

File: kornia/filters/unsharp.py (hunk 3 above)
Comment (edgarriba) [reply kornia/filters/unsharp.py:16]: @AnimeshMaheshwari22 For this things use `make autopep8` from root

File: kornia/filters/unsharp.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
+from typing import Tuple
+import numpy as np
+import torch
+import torch.nn as nn
+
+import kornia
+from kornia.filters import GaussianBlur2d
+
+def sharpen(
```
Comment (edgarriba) [inline kornia/filters/unsharp.py:9]: What should be the name of the functionality, sharpen, unsharpen?

File: kornia/filters/unsharp.py (hunk 5 above)
Comment (edgarriba) [reply kornia/filters/unsharp.py:35]: Right, make input/output tensor

File: kornia/filters/unsharp.py (hunk 6 above)
Comment (edgarriba) [reply kornia/filters/unsharp.py:36]: I would even use the functional for consistency `gaussian_blur2d`. The from the top import as `from kornia.filters import gaussian_blur2d`

File: kornia/filters/unsharp.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    data_blur: torch.tensor = gauss(data)
+    data_new = data + (data - data_blur)
+    img_new = kornia.tensor_to_image(data_new) * 255
+    return img_new
+    
+
+
+
+
+class unsharp_mask(nn.Module):
```
Comment (edgarriba) [inline kornia/filters/unsharp.py:46]: Name of the class should go in CamelCase style

File: test/filters/test_unsharp_mask.py (hunk 8 above)
Comment (edgarriba) [reply test/filters/test_unsharp_mask.py:10]: Agree

File: kornia/filters/unsharp.py (hunk 6 above)
Comment (AnimeshMaheshwari22) [reply kornia/filters/unsharp.py:36]: gauss is an operator that initializes the Gaussian Blur with parameters right? So do we need typing?

File: kornia/filters/unsharp.py (hunk 5 above)
Comment (edgarriba) [reply kornia/filters/unsharp.py:35]: what I'd suggest is the following:
```python
data_blur: torch.Tensor = gaussian_blurd2d(data, kernel_size, sigma)
out: torch.Tensor = data + (data - data_blur)
return out
```

Comment (AnimeshMaheshwari22) [reply kornia/filters/unsharp.py:35]: I have already implemented this:
    data=input.float()
    gauss= kornia.filters.GaussianBlur2d(kernel_size, sigma)
    data_blur: torch.Tensor = gauss(data)
    data_sharpened: torch.Tensor = data + (data - data_blur)
    return data_sharpened
I am working on the tutorial part. Will raise a PR soon

Comment (AnimeshMaheshwari22) [reply kornia/filters/unsharp.py:35]: Oh I understand. You suggest directly using gaussian_blur2d as you had stated earlier "I would even use the functional for consistency gaussian_blur2d. The from the top import as from kornia.filters import gaussian_blur2d" here
